spinalcordtoolbox/reports/qc2.py

```diff
@@ -538,16 +575,45 @@ def sct_deepseg_sagittal(
     ) if fname_seg_lesion else None
     if fname_seg_lesion:
         img_seg_lesion.data = (img_seg_lesion.data > 0.5) * 1
-
-    # The radius is set to display the entire slice
-    radius = (img_input.dim[1]//2, img_input.dim[2]//2)
+    img_qc_seg = resample_nib(
+        image=img_qc_seg,
+        image_dest=img_input,
+        interpolation='linear',
+    ) if fname_qc_seg else None
+
+    # Using -qc-seg mask if available, we remove slices which are empty (except for the first 3 and last 3 slices just around the segmented slices)
+    if fname_qc_seg:
+        # We dilate the mask to keep the first and last 2 slices
+        first_slice = min(np.where(img_qc_seg.data)[0]) - 2
+        last_slice = max(np.where(img_qc_seg.data)[0]) + 2
+        # Now we remove the slices from the input image, the spinal cord mask and the lesion mask
+        img_input.data = img_input.data[first_slice:last_slice]
+        img_seg_sc.data = img_seg_sc.data[first_slice:last_slice]
+        if img_seg_lesion:
+            img_seg_lesion.data = img_seg_lesion.data[first_slice:last_slice]
+        img_qc_seg.data = img_qc_seg.data[first_slice:last_slice]
 
     # Each slice is centered on the segmentation
     logger.info('Find the center of each slice')
-    centers = np.array([radius] * img_input.data.shape[0])
+    # Use the -qc-seg mask if available, otherwise use the spinal cord mask
+    if fname_qc_seg:
+        centers = np.array([center_of_mass(slice) for slice in img_qc_seg.data])
+    else:
+        centers = np.array([center_of_mass(slice) for slice in img_seg_sc.data])
     inf_nan_fill(centers[:, 0])
     inf_nan_fill(centers[:, 1])
 
+    # If -qc-seg is available, use it to generate the radius, otherwise, it is set to the standard value of (15,15)
+    # The height is always the entirety of the image height.
+    # The width is set to the maximum width of the spinal cord mask dilated by 20% or 1/2 of the image width, whichever is larger.
+    if fname_qc_seg:
+        widths = [np.max(np.where(slice)[1]) - np.min(np.where(slice)[1]) if np.sum(slice) > 0 else 0 for slice in img_qc_seg.data]
+        widths = [(w*1.2)//2 for w in widths]
+        height = np.floor(img_input.data.shape[2]/2).astype(int)
+        radius = (height, max(np.floor(img_input.data.shape[1]/4).astype(int), int(max(widths))))
+    else:
+        radius = (img_input.dim[1]//2, img_input.dim[2]//2)
```

Comment:
Should this be `(15, 15)` as in the previous review comment?

Reply:
No ! (15,15) only works on an axial view since it's the size of the spinal cord. On a sagittal view, I rather keep the full image.

Reply:
Ahhh of course, so I guess we just need to update the comments accordingly. :)